src/main/groovy/io/openliberty/tools/gradle/tasks/AbstractServerTask.groovy

```diff
@@ -284,20 +312,9 @@ abstract class AbstractServerTask extends AbstractTask {
         }
 
         // envProjectProps and serverEnvFile take precedence over server.env from configDirectory
+        // envCode goes here 
         File envFile = new File(serverDirectory, "server.env")
-        if ((server.env != null && !server.env.isEmpty()) || !envProjectProps.isEmpty()) {
-            if (serverEnvPath != null) {
-                logger.warn("The " + serverEnvPath + " file is overwritten by inlined configuration.")
-            }
-            writeServerEnvProperties(envFile, server.env, envProjectProps)
-            serverEnvPath = "inlined configuration"
-        } else if (server.serverEnvFile != null && server.serverEnvFile.exists()) {
-             if (serverEnvPath != null) {
-                logger.warn("The " + serverEnvPath + " file is overwritten by the " + server.serverEnvFile.getCanonicalPath() + " file.")
-            }
-            Files.copy(server.serverEnvFile.toPath(), envFile.toPath(), StandardCopyOption.REPLACE_EXISTING)
-            serverEnvPath = server.serverEnvFile.getCanonicalPath()
-        }
+        setServerEnv(envFile, serverEnvPath)
```

Comment:
I would call this `handleServerEnvFileAndProperties` since it is not actually setting something like a set method typically would. Also, it doesn't seem necessary to do line 316 here.